File: src/main/java/net/zoe/corporation/structures/commands/Command.java
package net.zoe.corporation.structures.commands;

import java.util.List;

public class Command {
   public String name;
   public List<String> aliases = List.of();
   public String description = "Not have a description proved";
   public String usage = "Not have a usage proved";
   public String category = "Miscellaneous";
   public boolean devsOnly = false;
   public boolean listInWebSite = true;
   public boolean enabled = true;
   public boolean hidden = false;
   public boolean canDisable = true;
   public int cooldown = 4;

   public void execute(CommandContext ctx) {
       return;
   };
}

File: src/main/kotlin/com/github/mrdroox/projects/zoe/structures/Command.kt
package com.github.mrdroox.projects.zoe.structures

@Suppress("LeakingThis")
open class Command {
    open var name: String? = null
    open var aliases: List<String> = listOf()
    open var description: String = "commands:${name}:description"
    open var usage: String = "commands:${name}:usage"
    open var category: String = "commands:${name}:category"
    open var devsOnly: Boolean = false
    open var listInWebSite: Boolean = true
    open var enabled: Boolean = true
    open var hidden: Boolean = false
    open var canDisable: Boolean = true
    open var cooldown: Int = 4

    open fun execute(ctx: CommandContext) {
        return
    }
}